syncer/syncer.go

```diff
@@ -1707,26 +1708,27 @@ func (s *Syncer) handleRowsEvent(ev *replication.RowsEvent, ec eventContext) err
 	return nil
 }
 
-func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) error {
+func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext, originSQL *string) error {
 	if bytes.Equal(ev.Query, []byte("BEGIN")) {
 		return nil
 	}
-	sql := strings.TrimSpace(string(ev.Query))
+	*originSQL = strings.TrimSpace(string(ev.Query))
```

Comment:
What about set originSQL to `strings.TrimSpace(string(ev.Query))` before we entering this function instead of using a pointer?